object-manipulation-2/src/defaults.ts

```diff
@@ -0,0 +1,15 @@
+/* exported defaults */
+function defaults(
+  target: Record<string, unknown>,
+  source: Record<string, unknown>
+): undefined {
+  const targetKeys = [];
+  for (const key in target) {
+    targetKeys.push(key);
+  }
+  for (const key in source) {
+    if (!targetKeys.includes(key)) {
+      target[key] = source[key];
+    }
+  }
+}
```

Comment:
We don't need this array to assign the properties of source to target. We only need to loop through one of the objects and add it's property and value to the other object if it does not exist.